doc/api/buffer.md

```diff
@@ -279,6 +279,101 @@ for (const b of buf) {
 Additionally, the [`buf.values()`][], [`buf.keys()`][], and
 [`buf.entries()`][] methods can be used to create iterators.
 
+## Class: `Blob`
+<!-- YAML
+added: REPLACEME
+-->
+
+A `Blob` encapsulates immutable, raw data.
+
+### `new buffer.Blob(sources[, options])`
+<!-- YAML
+added: REPLACEME
+-->
+
+* `sources` {string[]|ArrayBuffer[]|TypedArray[]|DataView[]|Blob[]} An array
+  of string, {ArrayBuffer}, {TypedArray}, {DataView}, or {Blob} objects, or
+  any mix of such objects, that will be stored within the `Blob`.
+* `options` {Object}
+  * `encoding` {string} The character encoding to use for string sources.
+    **Default**: `'utf8'`.
+  * `type` {string} The Blob content-type.
+
+Creates a new `Blob` object containing a concatenation of the given sources.
+The sources are copied into the 'Blob' and can therefore be safely modified
+after the 'Blob' is created.
+### `blob.arrayBuffer()`
+<!-- YAML
+added: REPLACEME
+-->
+
+* Returns: {Promise}
```

Comment:
I see this returns a `Promise`. Could we leverage this somehow and do the `memcpy` on a separate thread? Is it worth doing at all?

Reply:
I've been thinking about that. If the retained buffers are small the overhead of pushing it to another thread is going to be fairly significant and not really worth it. However, for large complex copies that involving coalescing multiple backing stores, then it's likely a really good idea to defer it. I'll play around with it a bit.